<!-- -*- mode: XML -*- -->

<launch>
  <arg name="file_name" default="20200402_TT_3F" />
  <arg name="fake_gps_file" default="$(arg file_name).gps" />
  <arg name="fake_gps_topic" default="/fake_gps/fix" />
  <arg name="pcd_file" default="20200402_3f.pcd " />

  <node pkg="autolabor_map_server" type="pcd_tf_finder" name="pcd_tf_finder_node" output="log">
    <param name="world_frame" type="str" value="world" />
    <param name="local_frame" type="str" value="local_map" />
    <param name="cloud_frame" type="str" value="map" />
    <param name="save_file"   type="str" value="$(find autolabor_map_server)/maps/$(arg file_name).yaml" />
    <param name="pcd_file"   type="str" value="$(arg pcd_file)" />
    <remap from="fake_gps" to="$(arg fake_gps_topic)" />
  </node>

  <node pkg="pcl_ros" type="pcd_to_pointcloud" name="pcd_map_publish" output="screen"
        args="$(env HOME)/pcd/$(arg pcd_file) 0.1">
    <param name="frame_id" value="map" />
    <remap from="cloud_pcd" to="points_map" />
  </node>

  <!-- Send a static GPS fix to every new subscriber. Edit latitude and longitude in launch/demo.gps to use your own position. -->
  <node pkg="rostopic" type="rostopic" name="fake_gps_fix"
        args="pub $(arg fake_gps_topic) sensor_msgs/NavSatFix
              --latch
              --file=$(find autolabor_sensors)/config/$(arg fake_gps_file)"
        output="screen"/>

  <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find autolabor_map_server)/rviz/map_tf_finder_3d.rviz" />

  <!-- <include file="$(find osm_cartography)/launch/viz_osm.launch">
    <arg name="url" value="package://autolabor_map_server/osm/taipei_test_area.osm" />
  </include> -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="world_to_earth_broadcaster"
        args="0 0 0 0 0 0 world earth" />

  <include file="$(find osm_cartography)/launch/viz_osm.launch">
    <arg name="url" value="package://autolabor_odom/osm/20200326_TaipeiTech.osm" />
  </include>

</launch>
